src/ImageSharp/Formats/Jpeg/Components/Encoder/Component.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Formats.Jpeg.Components.Encoder
+{
+    /// <summary>
+    /// Represents a single frame component.
+    /// </summary>
+    internal class Component : IDisposable
+    {
+        private readonly MemoryAllocator memoryAllocator;
+
+        public Component(MemoryAllocator memoryAllocator, int horizontalFactor, int verticalFactor, int quantizationTableIndex)
+        {
+            this.memoryAllocator = memoryAllocator;
+
+            this.HorizontalSamplingFactor = horizontalFactor;
+            this.VerticalSamplingFactor = verticalFactor;
+            this.SamplingFactors = new Size(this.HorizontalSamplingFactor, this.VerticalSamplingFactor);
```

Comment:
```suggestion
            this.SamplingFactors = new Size(horizontalFactor, verticalFactor);
```
I guess this produces better code, especially when the JIT has some work with inlining, etc. (to not reach some CSE limits and so on).